cmd/hostd/run.go

```diff
@@ -126,6 +126,37 @@ func startLocalhostListener(listenAddr string, log *zap.Logger) (l net.Listener,
 	return
 }
 
+func getRandomOpenPort() (uint16, error) {
+	l, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return 0, err
+	}
+	defer l.Close()
+	_, portStr, err := net.SplitHostPort(l.Addr().String())
+	if err != nil {
+		return 0, fmt.Errorf("failed to split port: %w", err)
+	}
+	port, err := strconv.ParseUint(portStr, 10, 16)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse port: %w", err)
+	}
+	return uint16(port), nil
+}
+
+func normalizeAddress(addr string) (string, error) {
```

Comment:
🤢

Reply:
Kind of sad, but a relatively straightforward way to solve the issue with the circular dependency on the RHP3 port, settings manager, and bandwidth limiters.

Open to ideas